src/django/pfb_analysis/models.py

```diff
@@ -58,10 +114,50 @@ def state(self):
         """
         return us.states.lookup(self.state_abbrev)
 
+    @classmethod
+    def name_for_label(cls, label):
+        return slugify(label)
+
     class Meta:
         unique_together = ('name', 'organization',)
 
 
+class AnalysisBatch(PFBModel):
+    """ Container for a grouping of AnalysisJobs that are run together
+
+    Allows us to track whether each job in a batch succeeded
+
+    An AnalysisJob does not need to belong to an AnalysisBatch
+
+    """
+    def __str__(self):
+        return '<AnalysisBatch: {} -- {}>'.format(str(self.uuid), self.created_at)
+
+    def submit(self):
+        """ Start all jobs in the batch """
+        for job in self.jobs.all():
+            job.run()
+
+    def cancel(self, reason=None):
+        """ Cancel all still-running jobs in the batch """
+        def chunks(l, n):
+            for i in range(0, len(l), n):
+                yield l[i:i + n]
+
+        client = boto3.client('batch')
+        if not reason:
+            reason = 'AnalysisBatch terminated by user at {}'.format(datetime.utcnow())
+
+        batch_job_ids = self.jobs.values_list('batch_job_id', flat=True)
+        for jobs in chunks(batch_job_ids, 100):
+            jobs_list = client.describe_jobs(jobs=jobs)['jobs']
+            for job in jobs_list:
+                if job['status'] in (JobState.SUBMITTED, JobState.PENDING, JobState.RUNNABLE,
+                                     JobState.STARTING, JobState.RUNNING,):
```

Comment:
Could add a convenience attribute like `JobState.ACTIVE_STATUSES` to capture this list, which would make it easier to adjust if AWS makes changes, especially if we end up writing these 5 out in more places.